### PR TITLE
 bpo-45026: More compact range iterator (alt)

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Keep the current and stop values instead of initial start, index and initial length.

https://bugs.python.org/issue45026

#89189

### CODE REVIEW

File: Objects/rangeobject.c [hunk 1]
```diff
@@ -766,27 +766,55 @@ PyTypeObject PyRange_Type = {
...
+        long    stop;
         long    step;
-        long    len;
 } rangeiterobject;
 
+/* Return number of items in range (lo, hi, step).  step != 0
+ * required.  The result always fits in an unsigned long.
+ */
+static unsigned long
+get_len_of_range(long lo, long hi, long step)
```
Comment (ambv) [inline Objects/rangeobject.c:760]: It would be a smaller change to just put a forward declaration here instead of moving the entire function around.

Comment (serhiy-storchaka) [reply Objects/rangeobject.c:760]: Yes, but in long term it is better location for this function. It does not make the change too large, I think.

I you prefer, I'll return it back and put a forward declaration.

**Comment by ambv** [comment]
The refleak here looks real:
```
test_range leaked [7, 7, 7] references, sum=21
test_range leaked [5, 5, 5] memory blocks, sum=15
```

**Comment by serhiy-storchaka** [comment]
@ambv please do not merge until I run benchmarks.

**Comment by serhiy-storchaka** [comment]
See also #27986.

**Pull request closed without merging**